tests/test_parser.py

```diff
@@ -32,7 +32,14 @@ def test_parser(infix, expected):
     ('a or b', ['c'], False),
     ('not a', ['b'], True),
     ('not a', ['a'], False),
-    ('not a', [], True)
+    ('not a', [], True),
+    ('not a(x)', [], True),
+    ('not a(x)', ['a(x)'], False),
+    ('a(x) or b', ['a(x)'], True),
+    ('a(x) or b', ['b'], True),
+    ('a(x) or b', [''], False),
+    ('a(x,y) or b', [''], False),
+    ('sometag(someValue,y) or b', [''], False)
```

Comment:
Can you add some more complex test cases in `test_complex_evaluation` ? Basically combining parenthesis usage with the *complex literals* that you've introduced?

Reply:
yep, added, let me know if i need to keep going :)